build-support/bin/generate_github_workflows.py

```diff
@@ -413,30 +410,18 @@ def bootstrap_caches(self) -> Sequence[Step]:
             },
         ]
 
-    def setup_primary_python(self, version: str | None = None) -> Sequence[Step]:
-        version = version or gha_expr("matrix.python-version")
+    def setup_primary_python(self) -> Sequence[Step]:
         ret = []
-        # We pre-install Pythons on our self-hosted platforms.
-        # We must set them up on Github-hosted platforms.
+        # We pre-install Python on our self-hosted platforms.
+        # We must set it up on Github-hosted platforms.
         if self.platform in GITHUB_HOSTED:
             ret.append(
                 {
-                    "name": f"Set up Python {version}",
+                    "name": f"Set up Python {PYTHON_VERSION}",
                     "uses": "actions/setup-python@v4",
-                    "with": {"python-version": version},
+                    "with": {"python-version": PYTHON_VERSION},
                 }
             )
-        ret.append(
-            {
-                "name": f"Tell Pants to use Python {version}",
-                "run": dedent(
-                    f"""\
-                    echo "PY=python{version}" >> $GITHUB_ENV
-                    echo "PANTS_PYTHON_INTERPRETER_CONSTRAINTS=['=={version}.*']" >> $GITHUB_ENV
-                    """
-                ),
-            }
-        )
         return ret
 
     def expose_all_pythons(self) -> Sequence[Step]:
```

Comment:
We still need this for tests of the python backend that use multiple interpreters.